web/src/pages/NewTransaction/Terms/Payment/GeneralTransaction/TokenAndAmount/TokenSelector/index.tsx

```diff
@@ -0,0 +1,234 @@
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  padding: 10px 16px;
+  cursor: pointer;
+  &:hover {
+    background: ${({ theme }) => theme.lightBlue};
+  }
+  ${({ selected, theme }) =>
+    selected &&
+    `
+    background: ${theme.mediumBlue};
+    border-left: 3px solid ${theme.primaryBlue};
+    padding-left: 13px;
+    &:hover {
+      background: ${theme.mediumBlue};
+    }
+  `}
+`;
+
+export const TokenLogo = styled.img`
+  width: 24px;
+  height: 24px;
+`;
+
+export const TokenLabel = styled.span`
+  color: ${({ theme }) => theme.primaryText};
+`;
+
+const DropdownContent = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+`;
+
+const StyledLogoSkeleton = styled(Skeleton)`
+  width: 22.5px;
+  height: 22.5px;
+  margin-left: 2px;
+  border-radius: 50%;
+`;
+
+const ReStyledModal = styled(StyledModal)`
+  display: flex;
+  width: 500px;
+`;
+
+const StyledSearchbar = styled(Searchbar)`
+  width: 100%;
+  input {
+    font-size: 16px;
+  }
+`;
+
+const ItemsContainer = styled.div`
+  display: flex;
+  width: 100%;
+  flex-direction: column;
+  margin-top: 24px;
+`;
+
+const StyledP = styled.p`
+  display: flex;
+  align-self: flex-start;
+  font-weight: 600;
+  margin: 0;
+  margin-bottom: 28px;
+`;
+
+const TokenSelector = () => {
+  const { address } = useAccount();
+  const { chain } = useNetwork();
+  const { sendingToken, setSendingToken } = useNewTransactionContext();
+  const [tokens, setTokens] = useLocalStorage("tokens", []);
+  const [filteredTokens, setFilteredTokens] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchQuery, setSearchQuery] = useState("");
+  const containerRef = useRef(null);
+  const [loading, setLoading] = useState(true);
+  const alchemyInstance = new Alchemy(alchemyConfig(chain?.id));
+  useClickAway(containerRef, () => setIsOpen(false));
+
+  useEffect(() => {
+    if (address && chain) {
+      initializeTokens(address, setTokens, setLoading, chain, alchemyInstance);
+    }
+  }, [address, chain]);
+
+  useEffect(() => {
+    if (tokens?.length > 0) {
+      const nativeToken = tokens.find((token) => token.address === "native");
+      setSendingToken(JSON.parse(localStorage.getItem("selectedToken")) || nativeToken);
+    }
+  }, [tokens, setSendingToken]);
+
+  const handleSelectToken = (token) => {
+    setSendingToken(token);
+    localStorage.setItem("selectedToken", JSON.stringify(token));
+    setIsOpen(false);
+  };
+
+  const handleSearch = async (query) => {
+    setSearchQuery(query);
+
+    if (!query) {
+      setFilteredTokens(tokens);
+      return;
+    }
+
+    const isAddress = query.startsWith("0x") && query.length === 42;
+    if (isAddress) {
+      try {
+        const metadata = await alchemyInstance.core.getTokenMetadata(query.toLowerCase());
+        const resultToken = {
+          symbol: metadata.symbol,
+          address: query.toLowerCase(),
+          logo: metadata.logo || "https://via.placeholder.com/24",
```

Comment:
- Maybe have the default / fallback logo as ETH or native chain currency.
- Can we abstract this logic to useERC20TokensSymbol hook? 🤔